services/core-api/app/api/incidents/models/mine_incident_category_xref.py

```diff
@@ -0,0 +1,43 @@
+from sqlalchemy.orm import validates
+from sqlalchemy.schema import FetchedValue
+from app.extensions import db
+from app.api.utils.models_mixins import Base
+
+
+class MineIncidentCategoryXref(Base):
+    __tablename__ = 'mine_incident_category_xref'
+    mine_incident_id = db.Column(
+        db.Integer,
+        db.ForeignKey('mine_incident.mine_incident_id'),
+        primary_key=True,
+        nullable=False)
+    mine_incident_category_code = db.Column(
+        db.String(3),
+        db.ForeignKey('mine_incident_category.mine_incident_category_code'),
+        primary_key=True,
+        nullable=False)
+
+    @classmethod
+    def create_batch(cls, mine_incident_id, mine_incident_category_codes, add_to_session=True):
+        if mine_incident_category_codes is not None:
+            mine_incident_category_codes.sort()
+            for mine_incident_category_code in mine_incident_category_codes:
+                if mine_incident_category_code is None:
+                    continue
+                new_mine_incident_category_code = MineIncidentCategoryXref.create(
+                    mine_incident_id, mine_incident_category_code, add_to_session)
+                new_mine_incident_category_code.save()
```

Comment:
Everything regarding the add_to_session, .save(), .save(commit=False) seems weird to me, as I think we just need to save() it, but I followed a previous pattern someone did when creating the code for creating recommendations.

Reply:
this seems unnecessary because a simple for loop in the resource can do this. We also want to avoid creating these relationships by explicitly referring to id's. 

eg. what if the mine_incident is brand new and hasn't gotten gotten an ID yet.

the consumer of this function should just loop like this eg from permit_amendment.py resource file. 

`        for newFile in uploadedFiles:
            new_pa_doc = PermitAmendmentDocument(
                document_name=newFile['fileName'],
                document_manager_guid=newFile['document_manager_guid'],
                mine_guid=permit.mine_guid,
            )
            new_pa.related_documents.append(new_pa_doc)
`
using the relationship with the secondary flag will let you refer to it as a set (and just append) then the ORM deals with the PK/FK's as it gets saved.

Reply:
I've tested creating brand new mine incidents and this doesn't seem to be a problem, but I'll check it out.

Reply:
hmm i see very similar assignments in the same POST, i'm surprised they work as they do. because until the object has been saved, the serial ID should still be null, maybe it's generated when it's attached to the session,,,